### ISSUE TITLE
Spruce doesn't work when loading with turbolinks

### ISSUE THREAD

**Issue by bdtomlin** [ryangjchandler/spruce#62]
Spruce does not work when using Turbolinks with the Alpine Turbolinks Adapter.

When clicking on a link handled by Turbolinks the following error occurs:

```
TypeError: t.subscribe is not a function. (In 't.subscribe(e)', 't.subscribe' is undefined)
```

With Turbolinks disabled everything works as expected. I'm new to Alpine and Spruce, so I'm not sure if this issue should be posted here or on the Turbolinks adapter repo. I thought I'd start here :)

**Comment by ryangjchandler** [comment]
@bdtomlin thanks for posting, I don't use Turbolinks myself so adding support in Spruce wasn't too high up on the priority list. This is hopefully something I can tackle in v2, I'll speak to @SimoTod too to see if there is a simpler way of adding Turbolinks support sooner.

**Comment by SimoTod** [comment]
👍I'll try to reproduce the error soon and see if it's something we can fix without major changes.

**Comment by bdtomlin** [comment]
Thanks for the quick responses. Please let me know if there's any way I can help.

**Comment by bdtomlin** [comment]
Some interesting additional information...

If I just remove the following line: https://github.com/ryangjchandler/spruce/blob/c82c7b28870ca46251804cff953aa18f3efc439c/src/index.js#L19

Everything seems to work as expected. I may be missing some case I'm not aware of though.

**Comment by ryangjchandler** [comment]
@bdtomlin That line of code was PR'd by somebody to add a basic level of Turbolinks support.

**Comment by bdtomlin** [comment]
@ryangjchandler that's weird because it seems to be the thing that breaks turbo links support 🤷 

I'm not sure what scenario would make it necessary. I've tried everything I can think of without that line and it all works.

If the store is `attached` on `domReady` it should stay loaded throughout turbolinks clicks. It seems like there is no reason to listen to `turbolinks:render` and attempt to attach again. Again... maybe I'm missing something.

**Comment by ryangjchandler** [comment]
@bdtomlin Yeah, like I said, I don't use Turbolinks myself so I'm not sure about compatibility stuff.

I'll try to fix compatibility in v2, which I'm aiming to release in the next couple of days 🤞

**Comment by bdtomlin** [comment]
@ryangjchandler that sounds great. Good luck on v2! Thanks for making this library, it definitely fills in some small gaps that I had with alpine.

**Comment by ryangjchandler** [comment]
> 
> 
> @ryangjchandler that sounds great. Good luck on v2! Thanks for making this library, it definitely fills in some small gaps that I had with alpine.

My absolute pleasure. I'd love to spend some more time on it but at the moment, I can't justify it against client work, etc, so it's mostly just maintenance and patches, but I've got some cool things coming in v2.

**Comment by SimoTod** [comment]
@bdtomlin I think the original intention was to cover the case where you load the website on a page that does not have spruce and you then navigate to a page having spruce.

I haven't tried yet, I should find some time tomorrow. It would be good to console log self inside attach() and before the error to understand why subscribe is undefined

**Comment by ryangjchandler** [comment]
@SimoTod Yeah, I think I'm going to add some sort of `booted` property in v2 as well that will be set once it has been booted. That could help in cases like you mention.

**Comment by bdtomlin** [comment]
@SimoTod, When the error happens self is the document element.

**Comment by SimoTod** [comment]
@bdtomlin yeah, binding the context would remove the error. However, it seems redundant as you suggested earlier. `attach()` defines the magic property which persist between page loads so there's no need to redefine it (I tested a few cases and nothing breaks).
I suspect that code was required in an early version of spruce released before Alpine started supporting magic properties which was working in a different way (hooking into each component and redefining the x-init property).
Maybe the original author (@jameslkingsley) can confirm.
If Ryan is happy with that, feel free to send a PR to remove that line.

**Comment by ryangjchandler** [comment]
@SimoTod Thanks for looking into this, appreciate it!

@bdtomlin You can send a PR in for this now so that I can release a patch version on 1.x, or wait until the weekend when I release 2.x 👀

**Comment by jameslkingsley** [comment]
I haven't used Spruce for a while so I'm out the loop on what's changed - but I think I remember the issue being navigating to a page with a Spruce component - which with the new magic properties in Alpine _shouldn't_ be an issue.

**Comment by bdtomlin** [comment]
@SimoTod,

Pull request sent. Thanks! 😄

**Comment by ryangjchandler** [comment]
Fixed in v1.1.2. Thanks guys! ❤️
